app/src/main/kotlin/at/bitfire/davdroid/ui/setup/LoginActivity.kt

```diff
@@ -71,7 +71,7 @@ class LoginActivity @Inject constructor(): AppCompatActivity() {
                 if (realScheme != null) {
                     val realUri = Uri.Builder()
                         .scheme(realScheme)
-                        .authority(uri.host)
```

Comment:
Very good that you found it!

A bit disappoint that we can't just use `.authority(uri.authority)` because of the `userInfo`, right?

Using the `host:port` explicitly feels a bit strange. And I wonder especially whether it would also add the standard port 443 to what the user sees? Because I think port 443 and port -1 ("default") are two different things.

If this is a thing, maybe we could use Java URI instead to process the incoming `intent.dataString` and then convert it to an Android Uri?

Reply:
Seems like I running into the same problem with Java URI. How about `.encodedAuthority(uri.host + if (uri.port > -1) ":"+uri.port else "")` ?

Reply:
Yeah, well if it works fine like that, we can use it. :+1: